src/deepsparse/server/build_logger.py

```diff
@@ -111,42 +110,90 @@ def build_leaf_loggers(
                     _build_function_logger(metric_function, target_identifier, loggers)
                 )
-    return function_loggers
-
-
-def _create_system_logger(loggers: Dict[str, BaseLogger]) -> FunctionLogger:
-    # returns a function logger that matches to all system logs, logging
-    # every system call to each leaf logger
-    return _build_function_logger(
-        metric_function_cfg=MetricFunctionConfig(
-            func="identity", frequency=1, target_loggers=None
-        ),
-        target_identifier=f"category:{MetricCategories.SYSTEM.value}",
-        loggers=loggers,
-    )
+    return data_loggers
+
+
+def build_system_loggers(
+    loggers: Dict[str, BaseLogger], system_logging_config: SystemLoggingConfig
+) -> List[FunctionLogger]:
+    """
+    Create a system loggers according to the configuration specified
+    in `system_logging_config`. System loggers are FunctionLogger instances
+    responsible for logging system groups metrics.
+
+    :param loggers: The created "leaf" loggers
+    :param system_logging_config: The system logging configuration
+    :return: A list of FunctionLogger instances responsible for logging system data
+    """
+    system_loggers = []
+    system_config_groups = extract_system_group_data(system_logging_config)
+    for config_group_name, config_group_args in system_config_groups.items():
+        system_loggers.append(
+            _build_function_logger(
+                metric_function_cfg=MetricFunctionConfig(
+                    func="identity",
+                    frequency=1,
+                    target_loggers=config_group_args.target_loggers,
+                ),
+                target_identifier=_get_target_identifier(
+                    config_group_name, is_system_logging_identifier=True
+                ),
+                loggers=loggers,
+            )
+        )
+    return system_loggers
+
+
+def extract_system_group_data(
+    system_logging_config: "SystemLoggingConfig",
+) -> Dict[str, SystemLoggingGroup]:
+    """
+    Extract the system logging groups data from the system logging configuration
+    :param system_logging_config: The system logging configuration
+    :return: A dictionary that contains a mapping from a system logging group name
+        to its arguments
+    """
+    if not system_logging_config.enable:
+        return {}
+    target_loggers = system_logging_config.target_loggers
+
+    # extract exclusively the system logging groups data
+    system_config_groups = {
+        config_group_name: config_group_args
+        for (config_group_name, config_group_args) in system_logging_config
+        if isinstance(config_group_args, SystemLoggingGroup)
+    }
+    # propagate the global target loggers information if applicable
+    for config_group_name, config_group_args in system_config_groups.items():
+        if target_loggers and not config_group_args.target_loggers:
+            config_group_args.target_loggers = target_loggers
```

Comment:
this code could read a bit better as a single for loop since we are creating a dictionary and then immediately mutating it